src/ResourceManagement/Batch/ApplicationPackagesImpl.cs

```diff
@@ -59,7 +59,7 @@ protected override IList<ApplicationPackageImpl> ListChildResources()
                     childResources.Add(new ApplicationPackageImpl(applicationPackage.Name, parent, applicationPackage, client));
                 }
             } catch (CloudException) {
-
+                //ignore the exception when creating application package during application definition stage
```

Comment:
What is the reason to ignore the exception?

Reply:
The exception will come when it calls batchAccount.Update().DefineNewApplication(appId).DefineNewApplicationPackage(appPackageName).Attach().Apply(). The request to list all application packages will get response of "the specific application does not exist" as we will commit and sync up both the application and application package at same time. What we have done on Java side is that we register error for CloudException so it return empty child resource list in the same situation. Thus, we need to do similar thing in .net side to catch the exception and return empty list.